lib/libmuser.c

```diff
@@ -1105,39 +1088,88 @@ free_sparse_mmap_areas(lm_reg_info_t *reg_info)
 }
 
 static int
-lm_caps_init(lm_ctx_t *lm_ctx, lm_cap_t *caps, int nr_caps)
+pci_config_setup(lm_ctx_t *lm_ctx, const lm_dev_info_t *dev_info)
 {
+    lm_reg_info_t *cfg_reg;
+    lm_reg_info_t zero_reg;
+    int i;
     int err;
 
-    assert(lm_ctx);
-    assert(caps);
+    assert(lm_ctx != NULL);
+    assert(dev_info != NULL);
 
-    if (!nr_caps) {
-        return 0;
+    // Convenience pointer.
+    cfg_reg = &lm_ctx->pci_info.reg_info[LM_DEV_CFG_REG_IDX];
+
+    // Set a default config region if none provided.
+    memset(&zero_reg, 0, sizeof(zero_reg));
+    if (memcmp(cfg_reg, &zero_reg, sizeof(*cfg_reg)) == 0) {
+        cfg_reg->flags = LM_REG_FLAG_RW;
+        cfg_reg->size = PCI_CFG_SPACE_SIZE;
```

Comment:
Missed `.fn = noop_cb`.

Reply:
Actually, I meant to take it out entirely. I'll fix it so that noop_cb actually goes for good. I added it as a hack to get the Piter demo ready, but in reality we don't need it. Libmuser needs to take care of everything config-space related, so let's focus on that. We should actually refuse a config access callback on dev_info if one is provided. On the config space, (I believe) we only need callbacks for the vendor capabilities.